[batch] test_callback is flaky

I see this happening quite a bit.

```
=================================== FAILURES ===================================
______________________________ Test.test_callback ______________________________

self = <test.test_batch.Test testMethod=test_callback>

    def test_callback(self):
        app = Flask('test-client')
    
        d = {}
    
        @app.route('/test', methods=['POST'])
        def test():
            d['status'] = request.get_json()
            return Response(status=200)
    
        server = ServerThread(app)
        try:
            server.start()
    
            j = self.batch.create_job(
                'alpine',
                ['echo', 'test'],
                attributes={'foo': 'bar'},
                callback=server.url_for('/test'))
            j.wait()
    
>           status = d['status']
E           KeyError: 'status'
```

--- a/batch/test/test_batch.py
+++ b/batch/test/test_batch.py
@@ -233,13 +233,19 @@ class Test(unittest.TestCase):
         server = ServerThread(app)
         try:
             server.start()
-
             j = self.batch.create_job(
                 'alpine',
                 ['echo', 'test'],
                 attributes={'foo': 'bar'},
                 callback=server.url_for('/test'))
             j.wait()
+
+            i = 0
+            while len(d) != 0:
+                time.sleep(0.100 * (3/2) ** i)
+                i += 1
+                if i > 14:
+                    break
 
             status = d['status']
             self.assertEqual(status['state'], 'Complete')

--- a/batch/test/test_dag.py
+++ b/batch/test/test_dag.py
@@ -255,7 +255,7 @@ def test_callback(client):
         i = 0
         while len(output) != 4:
             time.sleep(0.100 * (3/2) ** i)
-            i = i + 1
+            i += 1
             if i > 14:
                 break
         assert len(output) == 4

